src/Language/Haskell/Refact/API.hs

```diff
@@ -227,11 +234,48 @@ module Language.Haskell.Refact.API
  -- * from `Language.Haskell.Refact.Utils.ExactPrint'`
   , replace
   , setRefactAnns
+  , mergeRefactAnns
```

Comment:
Similar statements as per `fetchAnnsFinal`.  The `Anns` are a `Data.Map`, which has a `Monoid` instance, defining `mappend` as `Map.union`.

Also, `putRefactParsed` already does a merge with the new annotations.